embed/etcd.go

```diff
@@ -137,6 +139,11 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 	if err = e.serve(); err != nil {
 		return
 	}
+	e.Server.Cfg.OnShutdown = func() {
```

Comment:
I think this does not sync with `serve` routine? We populate `sctx.grpcServers`
in `(e *Etcd) serve()` (which calls `(sctx *serveCtx) serve` that creates `*grpc.Server`.
But `(e *Etcd) serve()` returns calling `(sctx *serveCtx) serve` in goroutines.